StorageCache 2023 01 01 API

rebecca337 opened the pull request:
## ARM API Information (Control Plane)

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow.
</i>
Azure 1st Party Service can try out the [Shift Left](https://aka.ms/ShiftLeft) experience to initiate API design review from ADO code repo.  If you are interested, may request engineering support by filling in with the form https://aka.ms/ShiftLeftSupportForm.

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. **End of January 2023**
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.  **End of January 2023**
  4. By default, Azure SDKs of all languages (.NET/Python/Java/JavaScript for both management-plane SDK and data-plane SDK, Go for management-plane SDK only ) MUST be refreshed with/after swagger of new version is published. If you prefer NOT to refresh any specific SDK language upon swagger updates in the current PR, please leave details with justification here.

### Contribution checklist (MS Employees Only):
- [x] I commit to follow the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist

> **Applicability**: :warning:
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs
> - Adding new properties
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits. You can use OpenAPIHub to initialize the PR for adding a new version. For more details refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them.

### Breaking Change Review Checklist
If you have any breaking changes as defined in the [Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy/), request approval from the Breaking Change Review Board.

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Additional details on the process and office hours are on the [Breaking Change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

NOTE: To update API(s) in public preview for over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

rebecca337 committed:
Adds base for updating Microsoft.StorageCache from version stable/2022-05-01 to version 2023-01-01

rebecca337 committed:
Updates readme

rebecca337 committed:
Updates API version in new specs and examples

rebecca337 committed:
2023-01-01 Swagger changes

rebecca337 committed:
Prettier fix

rebecca337 committed:
Fix miscellaneous typos

rebecca337 committed:
Add async op headers in examples

mentat9 commented:
>             "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"

v4 types are current and recommended. Comment applies throughout.

---
Refers to: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:530 in 03972d9. [](commit_id = 03972d9050f8c47f03d1dc733ef176c3f0ad898c, deletion_comment = False)

mentat9 commented:
>             "headers": {

You don't need to declare headers in the API spec. Comment applies throughout.

---
Refers to: specification/storagecache/resource-manager/Microsoft.StorageCache/stable/2023-01-01/storagecache.json:375 in 03972d9. [](commit_id = 03972d9050f8c47f03d1dc733ef176c3f0ad898c, deletion_comment = False)

mentat9 commented:
@rebecca337 - I didn't see any breaking changes so I went ahead with ARM review. You may need to investigate why breaking change review required label was applied.

@leni-msft, any idea? I also noticed the CI-MissingBaseCommit appears to be incorrectly applied, maybe related to the breaking change detection?

rebecca337 commented:
@mentat9 Regarding the comment about the "headers" in the API spec. We have "headers" tag in our previous API version that was approved. Are you saying this is incorrect? Do we remove all "headers" and the "location"/"azure-async-operation" tags?

mentat9 commented:
> @mentat9 Regarding the comment about the "headers" in the API spec. We have "headers" tag in our previous API version that was approved. Are you saying this is incorrect? Do we remove all "headers" and the "location"/"azure-async-operation" tags?

No, it's not incorrect. We have actually just been discussing what the guidance for header declarations should be. For now, we are recommending including it for async APIs as you have done (although it's not required), so feel free to ignore my comment.

leni-msft commented:
> @rebecca337 - I didn't see any breaking changes so I went ahead with ARM review. You may need to investigate why breaking change review required label was applied.
> 
> @leni-msft, any idea? I also noticed the CI-MissingBaseCommit appears to be incorrectly applied, maybe related to the breaking change detection?

The breaking change review required label was added because of the 2 errors detected in "Cross-Version breaking changes".

+@ruowan to help check why CI-MissingBaseCommit was added.

rebecca337 commented:
@mentat9 I will be deleting this PR as the automatic branch policies applied are preventing me from pushing any changes to this branch. I have opened this PR: https://github.com/Azure/azure-rest-api-specs/pull/22135 which addresses the comments you made.
